http/action/src/main/java/io/knotx/databridge/http/action/HttpAction.java

```diff
@@ -209,31 +272,45 @@ private MultiMap getFilteredHeaders(MultiMap headers, List<Pattern> allowedHeade
         .collect(MultiMapCollector.toMultiMap(o -> o, headers::getAll));
   }
 
-  private Single<FragmentResult> getFragmentResult(FragmentContext fragmentContext,
-      EndpointRequest endpointRequest, EndpointResponse endpointResponse) {
-    String transition = FragmentResult.ERROR_TRANSITION;
-
+  private Single<FragmentResult> createFragmentResult(FragmentContext fragmentContext,
+      EndpointRequest endpointRequest, EndpointResponse endpointResponse,
+      ActionLogger actionLogger) {
+    String transition = ERROR_TRANSITION;
     ActionRequest request = createActionRequest(endpointRequest);
     ActionPayload payload;
     if (SUCCESS.contains(endpointResponse.getStatusCode().code())) {
-      try {
-        payload = handleSuccessResponse(endpointResponse, request);
-        transition = FragmentResult.SUCCESS_TRANSITION;
-      } catch (DecodeException e) {
-        payload = handleInvalidResponseBodyFormat(request, e);
-      }
+      actionLogger.info(RAW_BODY, endpointResponse.getBody().toString());
+      payload = getActionPayloadOrHandleException(endpointRequest, endpointResponse, actionLogger,
+          request);
+      transition = FragmentResult.SUCCESS_TRANSITION;
     } else {
       payload = handleErrorResponse(request, endpointResponse.getStatusCode().toString(),
-          endpointResponse.getStatusMessage());
+          endpointResponse.getStatusMessage(), actionLogger);
       if (isTimeout(endpointResponse)) {
         transition = TIMEOUT_TRANSITION;
       }
     }
     updateResponseMetadata(endpointResponse, payload);
-
     Fragment fragment = fragmentContext.getFragment();
     fragment.appendPayload(actionAlias, payload.toJson());
-    return Single.just(new FragmentResult(fragment, transition));
+    return Single.just(new FragmentResult(fragment, transition, actionLogger.toLog().toJson()));
+  }
+
+  private ActionPayload getActionPayloadOrHandleException(EndpointRequest endpointRequest,
+      EndpointResponse endpointResponse, ActionLogger actionLogger, ActionRequest request) {
+    ActionPayload payload;
+    try {
+      payload = handleSuccessResponse(endpointResponse, request, actionLogger);
+    } catch (Exception e) {
+      logErrorAndRequest(actionLogger, e, endpointRequest);
+      logResponseOnErrorLevel(actionLogger, endpointRequest,
```

Comment:
```suggestion
      logResponseOnError(actionLogger, endpointRequest,
```
This name is more meaningful.